### ISSUE TITLE
String expressions don't interpret sequences of escaped control characters correctly

### ISSUE THREAD

**Issue by anthroid** [andmarti1424/sc-im#404]
String expressions don't preserve sequences of control character literals (`\` or `"`). It appears that either they are being re-evaluated again as special characters in the _result_ of an expression, rather than being treated as literals, or the expression parser is not handling the sequences with the right precedence when multiple escape characters are encountered.

Attempting several different methods to output the string literal `\"` in various forms always results in the expression being misinterpreted, even when double or triple escaped, concatenated, derived from another cell's value, or as the result of a function.

-|A|B|Expression|Expected result|Actual result
-|-|-|-|-|-
0|`\`|`"`|`\" A0 # B0 # "test"`|`\"test`|`"test`
1|`\\`|`"`|`\" A1 # B1 # "test"`|`\\"test`|`"test`
2|`\\\`|`"`|`\" A2 # B2 # "test"`|`\\\"test`|`"test`
3|`\`|`"`|`\" @sval("A",3) # @sval("B",3) # "test"`|`\"test`|`"test`
4|||`\" "\\\""`|`\"`|`"`
5|||`\" "\\\\\""`|`\\"`|`"`
6|||`\" "\\" # "\""`|`\"`|`" # `

In row `6` above, the first sequence `"\\"` should mean:
> "_Begin a string, escape one instance of_ `\`, _end the string_."

And the second sequence `"\""` should mean:
> "_Begin a string, escape one instance of_ `"`, _end the string_."

These would be concatenated by the `#` operator to produce `\"`.
Instead, it seems to be interpreted as:

1. Begin a string, escape `(null)`, skip and move forward _(the error might be here)_
2. Escape one instance of `"`, _treating the next characters as being "inside" the string_
3. Add the next `3` characters `space # space`
4. End the string
5. Encounter `\`, _(syntax error), end processing, drop remaining characters_

This appears to be confirmed by scim displaying the field's expression after pressing enter as:

`C5 (10 2 0) |"\\" # "`

Instead of the expected:
`C5 (10 2 0) ["\\"#"\""] |"\""`

It does not matter how many additional escapes are added, for example, all of the following result in the display value of `" #` and truncating of the expression:

`\" "\\\" # "\""`
`\" "\\\\" # "\""`
`\" "\\" # "\\""`

This might be caused by `printf`, where the __command-line__ implementation produces the following, which is identical to scim's output:

```
$ printf "\\"
\

$ printf "\""
"

# wrong
$ printf "\\\""
"

# right output, but requires double-escaping
$ printf "\\\\\""
\"

# right, using string literals by reference
$ printf "%s%s" "\\" "\""
\"
```

When using the __C function__, the results are as expected:

```c
printf("\\");
```
`\`

```c
printf("\"");
```
`"`

```c
printf("\\\"");    // correct
```
`\"`

```c
printf("%s%s", "\\", "\"");
```
`\"`

```c
printf("%c%c", '\\', '\"');
```
`\"`

Expressions in the attached file were entered as shown in the table above.
[escaped_control_chars.sc.txt](https://github.com/andmarti1424/sc-im/files/4568679/escaped_control_chars.sc.txt)

**Comment by andmarti1424** [comment]
You are probably right on these that should be fixed.

**Comment by andmarti1424** [comment]
@anthroid which command-line implementation are you refering to?

**Comment by anthroid** [comment]
@andmarti1424 Yes, the binary, e.g. `/usr/bin/printf` when run in bash, as compared to the output when calling `printf()` in C.
